From 7bffa89946b8eed8cb17e58666f8a9eb65888fc4 Mon Sep 17 00:00:00 2001
From: Bogdan Roman <bogdan-gabriel.roman@nxp.com>
Date: Thu, 16 Feb 2023 10:13:44 +0200
Subject: [PATCH 3/9] pkcs11-hse: add versioning for libpkcs-hse library

Conventionally, shared libraries should have version numbers. Add
version numbers for libpkcs-hse.
Also, add SONAME to libpkcs-hse. This is helpful for creating symlinks
in the rootfs.

Issue: ALB-9734
Upstream-Status: Pending 

Signed-off-by: Bogdan Roman <bogdan-gabriel.roman@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 Makefile | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index ebe5fa4..393875b 100644
--- a/Makefile
+++ b/Makefile
@@ -39,6 +39,9 @@ CFLAGS ?= -fPIC -Wall -g
 LDFLAGS ?=
 
 PKCS_LIB ?= libpkcs-hse.so
+PKCS_LIBVER_MAJOR = 1
+PKCS_LIBVER_MINOR = 0
+PKCS_LIBVER = $(PKCS_LIBVER_MAJOR).$(PKCS_LIBVER_MINOR)
 PKCS_SDIR = libpkcs
 PKCS_ODIR = $(PKCS_SDIR)/obj
 PKCS_SRCS = $(wildcard $(PKCS_SDIR)/*.c)
@@ -62,10 +65,11 @@ INCL = -I$(HSE_FWDIR)/interface                                                \
 
 DEFS := -DUIO_DEV=$(UIO_DEV)
 
-all: $(PKCS_LIB) examples
+all: $(PKCS_LIB).$(PKCS_LIBVER) examples
 
-$(PKCS_LIB): $(HSE_LIB).$(HSE_LIBVER) $(PKCS_OBJS)
-	$(CC) -shared $(CFLAGS) -L$(shell pwd) $(LDFLAGS) $(PKCS_OBJS) -o $@ -lhse
+$(PKCS_LIB).$(PKCS_LIBVER): $(HSE_LIB).$(HSE_LIBVER) $(PKCS_OBJS)
+	$(CC) -shared $(CFLAGS) -L$(shell pwd) -Wl,-soname,$(PKCS_LIB).$(PKCS_LIBVER_MAJOR) \
+	$(LDFLAGS) $(PKCS_OBJS) -o $@ -lhse
 
 $(PKCS_ODIR)/%.o: $(PKCS_SDIR)/%.c $(PKCS_ODIR)
 	$(CC) -c $(CFLAGS) $(INCL) $(LDFLAGS) $< -o $@
-- 
2.25.1

